### PR TITLE
Add automatic release on PyPI

### PR DESCRIPTION
**Pull request by mrbean-bremen**
- triggered on publishing a GH release
- includes creation of release documentation
- add PR templates, adapt issue templates
- minor documentation changes

### CODE REVIEW

File: .github/workflows/release-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Build package
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install wheel
+        cd main
+        echo "RELEASE_VERSION=$(cat pyfakefs/_version.py | cut -d "'" -f 2)" >> $GITHUB_ENV
+        python setup.py sdist bdist_wheel
```
Comment (nicoddemus) [inline .github/workflows/release-deploy.yml:32]: Modern practices suggest using `build`:

https://github.com/pytest-dev/pytest-mock/blob/b6ea52f32fd0221b0172dd46f3c8d9e04b139d35/.github/workflows/deploy.yml#L19-L25

File: .github/workflows/release-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    - name: Publish package to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (nicoddemus) [inline .github/workflows/release-deploy.yml:38]: If it is all the same to you, I suggest using `PYPI_TOKEN` for consistency with other repositories in pytest-dev (but this is not 100% consistent either, so feel free to disregard this).

File: .github/workflows/release-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    - name: Checkout gh-pages
+      uses: actions/checkout@v2
+      with:
+        ref: gh-pages
+        path: doc
+
+    - name: Copy and commit release documentation
+      run: |
+        cp -r main/gh-pages/* doc/release
```
Comment (nicoddemus) [inline .github/workflows/release-deploy.yml:56]: Ahh interesting, the docs are published in gh-pages.

In the future I suggest to use readthedocs.io, which builds the docs automatically for releases and even PRs for preview. 👍

**Review by nicoddemus — APPROVED** [review]

File: .github/workflows/release-deploy.yml (hunk 2 above)
Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:38]: Sure, will change this.

File: .github/workflows/release-deploy.yml (hunk 1 above)
Comment (nicoddemus) [inline .github/workflows/release-deploy.yml:32]:
```suggestion
        python -m pip install build
        cd main
        echo "RELEASE_VERSION=$(cat pyfakefs/_version.py | cut -d "'" -f 2)" >> $GITHUB_ENV
        python -m build
```

Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:32]: Thanks, that's better - will change this.

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:56]: Until now, I have always done it this way (including in `pytest-order`), but I will have a look at readthedocs in this case, thanks. Not for this release, though...

File: .github/workflows/release-deploy.yml (hunk 1 above)
Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:32]: Already had done this locally 😄

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:56]: The preview feature is indeed something I am missing in gh-pages, so I will definitely check this out!

Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:56]: Actually, setting up ReadTheDocs is very easy, as it turned out - I will probably change this before the release. Maybe tomorrow if I get to it... The PR preview is too good a feature to miss, and as far as I understand, there is literally nothing to do in the release workflow (apart from removing the GH Pages stuff again). Thanks for the tip!

Comment (nicoddemus) [reply .github/workflows/release-deploy.yml:56]: Sure thing!

The preview is not enabled by default, you need to go into advanced settings and enable it:

image

Comment (mrbean-bremen) [reply .github/workflows/release-deploy.yml:56]: Thanks, I noticed this. A bit odd that this cannot be configured in the configuration file...

**Pull request merged**